python/python/bystro/ancestry/train.py

```diff
@@ -136,12 +139,13 @@ def _parse_vcf_line_for_dosages(
     line: str, variants_to_keep: Container[Variant]
 ) -> tuple[Variant, list[int]] | None:
     # will throw ValueError if "PASS" not found, which is good
-    fields = line[: line.index("PASS")].split()
+    fields = line.split()
     variant = ":".join([fields[0], fields[1], fields[3], fields[4]])
-    assert_true("variant is a valid variant string", is_autosomal_variant(variant))
+    variant = variant if variant.startswith("chr") else "chr" + variant
+    assert_true(f"variant {variant} is a valid variant string", is_autosomal_variant(variant))
```

Comment:
Wondering whether we want to fold in is_autosomal_variant updates (can be a separate PR as well):

Wondering:
- Do we want to raise an assertion in this error or just return None? This function already has a way of communicating invalid sites: that is a None, so I'm just wondering if we need 2 mechanisms.

- We currently do a lot of processing on variant: splitting the line, rejoining the first field fields (skipping the id field) into a string, splitting it again in is_autosomal to check the components, again splitting it in _get_chr_from_variant, . Could you help me understand the benefit?
   - If we keep variant as a tuple (fields[0], fields[1], fields[3], fields[4]), or even better a named tuple, I think we get much better performance, with no downside. Pandas will happily take a tuple as a column name, since tuples are serializable.

Reply:
The short answer is that the autosomal variant check no longer actually make sense here, so the easiest thing to do is simply take it out.  (It made sense in the original context that the function was written, but now that we're applying it to parsing VCFs in the wild it's redundant: if a variant isn't one we want to keep, checking that against `variants_to_keep` is enough to determine that.

To the broader point, the benefit of having this check operate on strings was that strings are, for better or worse, currently the canonical representation of variants in this module.  It is not a great representation, for reasons discussed offline, and I'm happy to change it at some point: a namedtuple or the like is probably fine.  But, conceptually trivial as it is, it's major implementation change that will affect every part of the ancestry module, even affecting serialized model artifacts.  Like the old joke about finding half a worm in your apple, the only thing worse than a bad implementation is good one that half-replaces a bad one.  So if/when we change it I'd like to change it once and for all after we figure out exactly what it needs to handle: SNPs or arbitrary variants, genome build info, 0- or 1-based indexing, rsids, &c.

Most broadly speaking, I'd prefer to postpone this entire question of how quickly we're checking for the validity of a variant until we know that the ancestry service is basically functioning.  If we cut the time for this check down to zero we'd save approximately 1% of the total processing time, which we're doing in pure python to begin with.  As it is, we'd get a ~2x improvement over the regex implementation, so 0.5% overall.  (Recall that to check the validity of the `pos` value we either have to convert from str to int or handle the str with a regex anyway.). I thought we had consensus earlier that current VCF parsing performance was broadly acceptable, and that if we need generic performance, as opposed to relief from specifically targeted bottlenecks, then python was the wrong tool for the job to begin with and we should just rewrite (or better, find a library, cf. https://docs.rs/vcf/latest/vcf/)

The aim of this PR is really just to get ancestry back up and running so we can proceed with the end-to-end testing.  Once we're happy with what the model is giving us, I'm happy to make it go blazing fast.  But there are always going to be miscellaneous plumbing issues like the msgspec thing the first time we stand any new kind of service up and I'd like to try to see the end of the tunnel before optimizing.